websocket/src/websocket.cpp

```diff
@@ -234,7 +234,7 @@ static void DestroyConnection(WebsocketConnection* conn)
     }
 #else
     if (conn->m_Connection)
-        dmConnectionPool::Return(g_Websocket.m_Pool, conn->m_Connection);
+        dmConnectionPool::Close(g_Websocket.m_Pool, conn->m_Connection);
```

Comment:
I agree it's better to close it since we don't have any error result available here.

_Related code found in http_client.cpp: ~ Response(), where it checks the status before choosing to close or return the socket._